### ISSUE TITLE
WSL Performance Issues

### ISSUE THREAD

**Issue by BoscoDomingo** [usebruno/bruno#2524]
### I have checked the following:

- [X] I use the newest version of bruno.
- [X] I've searched existing issues and found nothing related to my issue.

### Describe the bug

Using Bruno (installed on Windows) with collections stored in WSL leads to constant hang-ups, and many issues such as not being able to rename files #2269 #2342 , or not being able to send requests altogether since. Other programs work just fine, so it's definitely something with Bruno.

### Screenshots/Live demo link

An example of the program not allowing me to cancel a request (which had been sent 5 minutes earlier, and took a couple minutes to even be sent in the first place)

https://github.com/usebruno/bruno/assets/46006784/2af33784-2850-4bc2-a0f7-657bee581286

**Comment by krummbar** [comment]
Since you're mentioning golden edition. Did you try disabling the git Integration Feature?
I've observed some performance issues with it being active and having a couple of collections in place. If this improves things this may be a good indicator of the underlying issue.

**Comment by BoscoDomingo** [comment]
> Since you're mentioning golden edition. Did you try disabling the git Integration Feature? I've observed some performance issues with it being active and having a couple of collections in place. If this improves things this may be a good indicator of the underlying issue.

@krummbar That seems to work. I've also disabled API Spec since I didn't use it, but I'll give this a try. If I don't report back, assume the issues are gone and Git integration is the culprit

**Comment by krummbar** [comment]
It's just a blind guess though. Reporting back would be beneficial in any case. Imho if the Problem is related to git, Performance should bei improved. Specially since IT is a payed feature

**Comment by BoscoDomingo** [comment]
@krummbar Even with Git disabled, program still hangs:
![image](https://github.com/usebruno/bruno/assets/46006784/402467c9-1b2c-41c1-86db-9e3dd90b8eb7)

**Comment by Its-treason** [comment]
I suspect that read/write operations on WSL are sometimes kinda slow. This should normally not freeze the UI, but Bruno mostly uses synchronous operations  (`fs.writeSync`) instead of asynchronous  (Everything from `fs/promises`).

I add this to my to-do list and test this the next few days.

**Comment by Its-treason** [comment]
I investigated this a bit. I moved a collection with 30~ requests into a WSL folder, but didn't really see any freezes. To be sure i also moved a collection with ~2k requests to WSL, but i only had the normal ~5 seconds lags when interacting with the big collection, but that did cause any "window not responding".

Could you share more information about your system?

Also if you're feeling confident with debbugging electron, you can attach a debugger like describted here: https://www.electronjs.org/docs/latest/tutorial/debugging-main-process and do an performance recording.

**Comment by BoscoDomingo** [comment]
@Its-treason Hi, sure I can. Do you need hardware specs, OS version...? And I could try to debug Electron, although I just came back from holidays and this is for my work machine, so it may take a while before I can take a dig at it 👍🏼

**Comment by BoscoDomingo** [comment]
@Its-treason Still a problem in 1.25.0. I even deactivated my Gold license to see if that was part of the issue but to no avail. I'll try a performance recording to see if that helps. I've never worked with Electron but I assume it's just cloning the repo and running it, correct?

**Comment by Its-treason** [comment]
Hey @BoscoDomingo, thank you for taking the time and doing some performance recordings. You can either message me on Discord: https://discord.com/users/139058134596583424 Or email me `timon.sch71@gmail.com`. I don't have much time today, so I can first look the recordings tomorrow.

**Comment by BoscoDomingo** [comment]
@Its-treason Shooting an email 👍🏼

**Comment by Its-treason** [comment]
Hey @BoscoDomingo, thank you for the recording! I've had a look into your recordings, and it's quite interesting, because it has nothing to do with IO but, it's just the .bru file parser that is really slow. 

This was already noted here: https://github.com/usebruno/bruno/issues/2416 and it just seems that parsing files with 1–2 MB in size takes an about ~10 seconds of parsing. Because parsing is done in sync inside the main thread, this also blocks the render thread, resulting in a window not responding message.

What interests me now: How big are your .bru files? There was one outlier in the CPU Profile that took 2 seconds for parsing, so that file should have been around 200 KB - 400 KB in size. If all your files are small (under 10 KB) there is something other that causes the parser to perform even worse.

**Comment by BoscoDomingo** [comment]
@Its-treason Interesting find for sure. I've checked and the largest .bru files in my system are 12KB, so there's definitely something up with the parser:
![image](https://github.com/user-attachments/assets/bdc81222-d6a7-445c-81ba-92d15dc697be)

**Comment by BoscoDomingo** [comment]
@Its-treason Any updates on this front?

**Comment by Its-treason** [comment]
> @Its-treason Any updates on this front?

Not really, it's most likely some weird issue with the parser, I couldn't really figure it out. Anoop announced, that Bruno will move to YAML, which may fix your Problem.

**Comment by BoscoDomingo** [comment]
> > @Its-treason Any updates on this front?
> 
> Not really, it's most likely some weird issue with the parser, I couldn't really figure it out. Anoop announced, that Bruno will move to YAML, which may fix your Problem.

Oh nice! Any ideas when that might happen?

**Comment by Its-treason** [comment]
> > > @Its-treason Any updates on this front?
> > 
> > Not really, it's most likely some weird issue with the parser, I couldn't really figure it out. Anoop announced, that Bruno will move to YAML, which may fix your Problem.
> 
> Oh nice! Any ideas when that might happen?

Snoop said something about this quarter, in the last community call. I hope pretty soon.

**Comment by MichaKnoe** [comment]
We recognized that the performance problem increases with the count of imported collections. It makes WSL almost unsuable having Bruno UI open. On my system the critical mass is at 5 to 8 collections. All of those collections have there bru files located in WSL.

**Comment by BoscoDomingo** [comment]
Recently (version 1.35 IIRC) the problems mentioned in this issue seem to have gone away. I'm pretty sure it may have something to do with #3482 but I'm not certain. Out of precaution I haven't updated since, and it's all worked as expected

**Comment by robinbertram** [comment]
I'm experiencing the same issue. High CPU usage for both Bruno and vmmemWSL on Windows 11 with a collection of 700 requests.

When I move the folder to Desktop from a folder inside my WSL environment Bruno goes down to 0-0.1% and vmmemWSL goes down to 1-2%.

I'm running Bruno 2.1.0.

**Comment by mhouel-coexya** [comment]
Exactly the same behavior with version **2.10.0**.
I'm using a laptop with **Windows 11** and **WSL** 2.5.9 (Ubuntu 24.04).

I open Bruno on Windows and the files are stored in the WSL (\\wsl.localhost\Ubuntu\home\user\...).
I open a collection of ~100 requests.
When the collection is opened, **Bruno and vmmemWSL use 90% CPU constantly** (with the fans waking up...).
When I close it, vmmemWSL goes back to ~2% CPU.

Now, I moved my working folder out of the WSL, and I have no more performance issues. 🥇

**Comment by bramvdh** [comment]
Any updates on this yet, this is really annoying if you want to change any env vars, bruno becomes unusable tbh

**Comment by Pragadesh44-Bruno** [comment]
@bramvdh Thanks for reporting this! Could you please test our nightly version at https://github.com/usebruno/bruno-nightly-builds and check if you’re still experiencing the same problem?

**Comment by bramvdh** [comment]
> [@bramvdh](https://github.com/bramvdh) Thanks for reporting this! Could you please test our nightly version at https://github.com/usebruno/bruno-nightly-builds and check if you’re still experiencing the same problem?

will give that a try

**Comment by bramvdh** [comment]
@Pragadesh44-Bruno seems to be better indeed. thx a bunch

**Comment by asanglard** [comment]
Hello,

I have the same issue here, when Bruno UI is opened, it burns a lot of CPU related to the init process.

<img width="900" height="60" alt="Image" src="https://github.com/user-attachments/assets/77fe56d5-51a6-42d0-9b6a-9cc008ca1b33" />

it makes hard to use Bruno in our context and it is not very eco-friendly...

the issue is still present with the lastest nightly build mentioned above

**Comment by Pragadesh44-Bruno** [comment]
@asanglard Could you please share a video of this? It would be easier to identify the process causing the issue from a video, and the terminal logs could be very helpful as well. The performance fix for the environments modal should be working now.

**Comment by JoshuaHintze** [comment]
I know that some of the fixes recently in bruno 2.14.0 like this

`Reduced high CPU usage caused by WMI queries`

Seemed great at first, but now that I'm on 2.15.1 it feels like we have regressed again. CPU usage of bruno sitting idle with installation Windows and collections on WSL now eat about 5-10% of CPU usage all the time. Anyone else seeing this?

**Comment by claudioweiler** [comment]
Just some 50 cents.

Windows Apps reading on WSL mounts, as also the inverse, will performe HORRIBLE. Do not do this! Either use Windows, or WSL.

This is a known "issue" with disk operation conversions: https://learn.microsoft.com/en-us/windows/wsl/setup/environment#file-storage

ps. VS Code (on Windows) works well with WSL because it uses a integration bridge
